huaweicloud/services/ecs/resource_huaweicloud_compute_auto_launch_group.go

```diff
@@ -0,0 +1,469 @@
+	}
+
+	id, err := jmespath.Search("auto_launch_group_id", createAutoLaunchGroupRespBody)
+	if err != nil {
+		return diag.Errorf("error creating auto launch group: %s is not found in API response", "id")
+	}
+	d.SetId(id.(string))
+
+	return resourceAutoLaunchGroupRead(ctx, d, meta)
+}
+
+func buildCreateAutoLaunchGroupBodyParams(d *schema.ResourceData, region string) map[string]interface{} {
+	bodyParams := map[string]interface{}{
+		"name":                               d.Get("name"),
+		"target_capacity":                    d.Get("target_capacity"),
+		"stable_capacity":                    utils.ValueIngoreEmpty(d.Get("stable_capacity")),
+		"excess_fulfilled_capacity_behavior": utils.ValueIngoreEmpty(d.Get("excess_fulfilled_capacity_behavior")),
+		"type":                               utils.ValueIngoreEmpty(d.Get("type")),
+		"instances_behavior_with_expiration": utils.ValueIngoreEmpty(d.Get("instances_behavior_with_expiration")),
+		"valid_since":                        utils.ValueIngoreEmpty(d.Get("valid_since")),
+		"valid_until":                        utils.ValueIngoreEmpty(d.Get("valid_until")),
+		"allocation_strategy":                utils.ValueIngoreEmpty(d.Get("allocation_strategy")),
+		"supply_option":                      utils.ValueIngoreEmpty(d.Get("supply_option")),
+		"spot_price":                         utils.ValueIngoreEmpty(d.Get("spot_price")),
+		"region_specs":                       buildAutoLaunchGroupRequestBodyRegionSpecs(d, region),
+	}
+	return bodyParams
+}
+
+func buildAutoLaunchGroupRequestBodyRegionSpecs(d *schema.ResourceData, region string) []map[string]interface{} {
+	regionSpecs := make([]map[string]interface{}, 0)
+	params := map[string]interface{}{
+		"region_id":              region,
+		"expect_target_capacity": d.Get("target_capacity"),
+		"expect_stable_capacity": d.Get("stable_capacity"),
+		"launch_template_config": map[string]interface{}{
+			"launch_template": map[string]interface{}{
+				"launch_template_id": d.Get("launch_template_id"),
+				"version":            d.Get("launch_template_version"),
+			},
+			"overrides": buildAutoLaunchGroupRequestBodyRegionSpecsOverrides(d.Get("overrides").(*schema.Set)),
+		},
+	}
+	regionSpecs = append(regionSpecs, params)
+	return regionSpecs
+}
+
+func buildAutoLaunchGroupRequestBodyRegionSpecsOverrides(rawParams *schema.Set) []map[string]interface{} {
+	if rawParams.Len() == 0 {
+		return nil
+	}
+	overrides := make([]map[string]interface{}, rawParams.Len())
+	for _, val := range rawParams.List() {
+		raw := val.(map[string]interface{})
+		params := map[string]interface{}{
+			"availability_zone_id": raw["availability_zone"],
+			"flavor_id":            raw["flavor_id"],
+			"spot_price":           utils.ValueIngoreEmpty(raw["spot_price"]),
+			"priority":             utils.ValueIngoreEmpty(raw["priority"]),
+			"weighted_capacity":    utils.ValueIngoreEmpty(raw["weighted_capacity"]),
+		}
+		overrides = append(overrides, params)
```

Comment:
the init length of `overrides` is not zero, so `append` is not correct
```suggestion
		overrides[i] = params
```